server/src/main/java/org/elasticsearch/action/admin/indices/mapping/get/TransportGetMappingsAction.java

```diff
@@ -45,26 +56,39 @@ public TransportGetMappingsAction(
     ) {
         super(
             GetMappingsAction.NAME,
-            transportService,
-            clusterService,
-            threadPool,
             actionFilters,
-            GetMappingsRequest::new,
-            indexNameExpressionResolver,
-            GetMappingsResponse::new
+            transportService.getTaskManager(),
+            clusterService,
+            threadPool.executor(ThreadPool.Names.MANAGEMENT)
         );
         this.indicesService = indicesService;
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+
+        transportService.registerRequestHandler(
+            actionName,
+            executor,
+            false,
+            true,
+            GetMappingsRequest::new,
+            (request, channel, task) -> executeDirect(task, request, new ChannelActionListener<>(channel))
+        );
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(GetMappingsRequest request, ClusterState state) {
+        return state.blocks()
+            .indicesBlockedException(ClusterBlockLevel.METADATA_READ, indexNameExpressionResolver.concreteIndexNames(state, request));
     }
 
     @Override
-    protected void doMasterOperation(
+    protected void localClusterStateOperation(
```

Comment:
This one is missing a call to `((CancellableTask) task).ensureNotCancelled();` somewhere

Reply:
I was viewing the `((CancellableTask) task).ensureNotCancelled()` call as a way to do a check prior to some (semi-)expensive work, to avoid doing it if it was going to be thrown away anyway. It seems like it would still have some value there, though only a little, as you say.

I'm fine either way, so I'll leave it to you whether to include them or not.